core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,8 +62,14 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // visible for testing
+  private[spark] val isPartitionEvalParallel: Boolean =
+    rdds.length > conf.getInt("spark.rdd.parallelListingThreshold", 10)
```

Comment:
@srowen, I don't think `enableParallelListing` was the option. This always used `parallelListingThreshold`, from the [original version](https://github.com/apache/spark/pull/8512/files#diff-eaababfc87ea4949f97860e8b89b7586). I'm happy to change it to the boolean version if there's consensus around that.